src/main/scala/io/iohk/ethereum/ledger/Ledger.scala

```diff
@@ -70,38 +96,46 @@ class Ledger(vm: VM) extends Logger {
               logs = logs
             )
 
-            BlockResult(newWorld, receipt.cumulativeGasUsed, receipts :+ receipt)
+            executeTransactions(otherStxs, newWorld, blockHeader, receipt.cumulativeGasUsed, acumReceipts :+ receipt)
+          case Left(error) => Left(TxsExecutionError(error))
         }
     }
-  }
 
   private[ledger] def executeTransaction(stx: SignedTransaction, blockHeader: BlockHeader, world: InMemoryWorldStateProxy): TxResult = {
     val gasPrice = UInt256(stx.tx.gasPrice)
     val gasLimit = UInt256(stx.tx.gasLimit)
 
-    val world1 = updateSenderAccountBeforeExecution(stx, world)
-    val result = runVM(stx, blockHeader, world1)
+    val worldBeforeTransfer = updateSenderAccountBeforeExecution(stx, world)
+    val result = runVM(stx, blockHeader, worldBeforeTransfer)
+
+    //FIXME: This only implements error handling as done in Homestead
```

Comment:
We should create an empty contract in case an `OutOfGas` exception is produced in `saveNewContract` (as mentioned in https://github.com/ethereum/EIPs/blob/master/EIPS/eip-2.mediawiki note 3.). This currently causes tx execution to fail in block 49018.